channel.go

```diff
@@ -1290,6 +1290,12 @@ func (ch *Channel) ExchangeUnbind(destination, key, source string, noWait bool,
 	)
 }
 
+// GetNextPublishSeqNo returns the sequence number of the next message to be
+// published, when in confirm mode.
+func (ch *Channel) GetNextPublishSeqNo() uint64 {
+	return ch.confirms.published
```

Comment:
```suggestion
func (ch *Channel) GetNextPublishSeqNo() uint64 {
	c.m.Lock()
	defer c.m.Unlock()
	return ch.confirms.published
```

doesn't the lock needs to be hold?

Reply:
`ch.confirms.Lock()` and `defer ch.confirms.Unlock()` would work better yet, or add a function in confirm?

Reply:
Yes, it makes sense to add `Lock` and `Unlock` for confirm. though it is Read operation, currently we are using `sync.Mutex` not `RWMutex` https://github.com/streadway/amqp/blob/master/confirms.go#L7 but I change it to `RWMutex` which is not going to impact any current functionality.

```
// Lock acquire the lock on confirms.
func (c *confirms) Lock() {
	c.m.Lock()
}

// Unlock release the locks on confirms.
func (c *confirms) Unlock() {
	c.m.Unlock()
}

Reply:
I would keep the mutex. It should be faster when the lock is only hold very short timespans, which is the case here.
It might be much slower with an rwmutex  (https://github.com/golang/go/issues/38813)